profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/CrashReportingWindows.cpp

```diff
@@ -0,0 +1,299 @@
+            SYM_STKWALK_DEFAULT))
+        {
+            auto module = FindModule(nativeStackFrame.AddrPC.Offset);
+
+            StackFrame stackFrame;
+            stackFrame.ip = nativeStackFrame.AddrPC.Offset;
+            stackFrame.sp = nativeStackFrame.AddrStack.Offset;
+            stackFrame.isSuspicious = false;
+            stackFrame.moduleAddress = module.second;
+            stackFrame.symbolAddress = nativeStackFrame.AddrPC.Offset;
+
+            std::ostringstream methodName;
+            methodName << module.first << "!<unknown>+" << std::hex << (nativeStackFrame.AddrPC.Offset - module.second);
+            stackFrame.method = methodName.str();
+
+            std::filesystem::path modulePath(module.first);
+
+            if (modulePath.has_filename())
+            {
+                const auto moduleFilename = modulePath.stem().string();
+
+                if (moduleFilename.rfind("Datadog", 0) == 0
+                    || moduleFilename == "libdatadog"
+                    || moduleFilename == "datadog"
+                    || moduleFilename == "libddwaf"
+                    || moduleFilename == "ddwaf")
+                {
+                    stackFrame.isSuspicious = true;
+                }
+            }
+
+            frames.push_back(std::move(stackFrame));
+        }
+    }
+
+    CloseHandle(thread);
+
+    return MergeFrames(frames, managedFrames);
+}
+
+std::string CrashReportingWindows::GetSignalInfo(int32_t signal)
+{
+    return std::string();
+}
+
+std::vector<ModuleInfo> CrashReportingWindows::GetModules()
+{
+    std::vector<ModuleInfo> modules;
+
+    HMODULE hModules[1024];
+    DWORD cbNeeded;
+    if (EnumProcessModules(_process, hModules, sizeof(hModules), &cbNeeded))
+    {
+        for (unsigned int i = 0; i < (cbNeeded / sizeof(HMODULE)); ++i)
+        {
+            MODULEINFO moduleInfo = {};
+            if (GetModuleInformation(_process, hModules[i], &moduleInfo, sizeof(moduleInfo)))
+            {
+                std::string resolvedModuleName = "<unknown>";
+
+                char moduleName[MAX_PATH];
+                if (GetModuleFileNameExA(_process, hModules[i], moduleName, sizeof(moduleName)))
+                {
+                    resolvedModuleName = moduleName;
+                }
+
+                modules.push_back({ (uintptr_t)moduleInfo.lpBaseOfDll, (uintptr_t)moduleInfo.lpBaseOfDll + moduleInfo.SizeOfImage, resolvedModuleName });
+            }
+        }
+    }
+
+    return modules;
+}
+
+std::pair<std::string, uintptr_t> CrashReportingWindows::FindModule(uintptr_t ip)
+{
+    for (auto& module : _modules)
+    {
+        if (ip >= module.startAddress && ip < module.endAddress)
+        {
+            return std::make_pair(module.path, module.startAddress);
+        }
+    }
+
+    return std::make_pair("", 0);
+}
+
+std::vector<StackFrame> CrashReportingWindows::MergeFrames(const std::vector<StackFrame>& nativeFrames, const std::vector<StackFrame>& managedFrames)
```

Comment:
This method could be moved as a protected member of `CrashReporting` class no?

Reply:
Right